### DISCUSSION TITLE
Words filter: Letter sequence and/or regex

### DISCUSSION THREAD

**Discussion by rusproject** [monkeytypegame/monkeytype#5964] [Ideas] [answered]
Currently, in custom mode, we can filter word lists by letters (specifying which letters words should or should not contain), but we can’t filter by specific letter sequences, like "eq." (or it's not clear how to do that?) 

It would be useful to add the ability to filter words by more than one consecutive letter, allowing users to target specific combinations.

Additionally, implementing regex filtering would be a powerful way to achieve this and much more. For example, if you wanted to practice words containing multiple challenging letters (like some of q, z, x, and p in one word, without specific order), a regex filter would make that possible. This feature would give us ultimate flexibility to create custom word lists for any needs.

**Comment by Miodec** [comment, marked as answer]
Just insert `eq` into the "include" field.

**Comment by alexchexes** [comment]
@Miodec
Pardon me, you’re right, it works. I tried before but didn’t get any results, probably due to the wrong language selection.

That said, this request is mostly about regex filtering. Could you comment on that?

**Reply by Miodec** [reply]
I guess i could add a regex option, but im not sure about the security of evaluating a user provided regex. I might have a look. Would a regex input replacing the 'include' and 'exclude' fields be enough?

**Reply by alexchexes** [reply]
@Miodec Yeah, I guess since we can negate most expressions within the regex itself, there’s probably no need for separate “include regex” and “exclude regex” fields. However, if you were to introduce a “Use regex” checkbox near the current include/exclude fields, it might make sense to have that option for both fields.

**Reply by Miodec** [reply]
I would probably just have a button group "simple" and "advanced" with advanced just replacing the length and include/exclude fields with one regex input.

**Reply by alexchexes** [reply]
Ok, regarding the security concerns, is the word filtering client-side? If so, I suppose there wouldn't be any issues. If it’s server-side, it might be a good idea to look into the "ReDoS" topic just in case.
